### PR TITLE
[ZEPPELIN-1346] Add contribution guidelines to the website

### PR DESCRIPTION
**Pull request by corneadoug**
### What is this PR for?
Recreation of #1354 
This PR is bringing Contribution guidelines to the website.
We also introduce a sidemenu on those pages allowing to create additional content.
As an example, for the Web Application Guidelines, we added a page about "Defining Components"

In a different PR, we will remove the contributing markdown, and change the links in the repository to the website.

The layout of the doc was slightly modified to allow a larger documentation at max size.
It was also modified to accommodate better on smaller screens.

### What type of PR is it?
Documentation

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1346

### How should this be tested?
Build the website locally and visit the community page

### Screenshots (if appropriate)
screen shot 2016-08-23 at 3 50 41 pm

screen shot 2016-08-31 at 10 33 36 am

#### Layout update (Before / After)

Navbar Bug fix

screen shot 2016-08-31 at 10 35 14 am

screen shot 2016-08-31 at 10 31 18 am

Small Device Margin Improvement

screen shot 2016-08-31 at 10 35 34 am

screen shot 2016-08-31 at 10 31 28 am

Content Width Change

screen shot 2016-08-31 at 10 33 50 am

screen shot 2016-08-31 at 10 32 12 am

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

**Comment by corneadoug** [comment]
@AhyoungRyu Sorry you will have to re-do your PR here :(

**Comment by AhyoungRyu** [comment]
@corneadoug Don't worry about that. Let me repush again :)

**Comment by corneadoug** [comment]
Alright LGTM

**Comment by AhyoungRyu** [comment]
good to go now 👯

**Comment by Leemoonsoo** [comment]
I've tested and it looks nice.

Some small feedback about sidemenu

- When sidemenu is displayed, content area become little bit too narrow.
- Items in Sidemenu is in consistent. i.e.

page `/contribution/general.html`, sidemenu
  - Documentation (/contribution/documentation.html)
  - Webapplication (/contribution/webapplication.html)

page `/contribution/webapplication.html`
  - Defining Components (/contribution/zeppelinweb/goodPracticeGuide01.html)

page `/contribution/zeppelinweb/goodPracticeGuide01.html`
  - Defining Components (/contribution/zeppelinweb/goodPracticeGuide01.html)

I think it'll better it sidemenu decide only one behavior either doesn't include current page, (like `general.html`,`webapplication.html`) or does include current page (like goodPracticeGuide01.html)

File: contribution/documentation.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+ - `{ASSET_PATH}/js/`: extra `.js` files can be placed
+ - `Gemfile`: defines bundle dependencies. They will be installed by `bundle install`.
+ - `Gemfile.lock`: when you run `bundle install`, bundler will persist all gems name and their version to this file. For the more details, see [Bundle "The Gemfile Lock"](http://bundler.io/v1.10/man/bundle-install.1.html#THE-GEMFILE-LOCK)
+ - `documentation_group`: `development/`, `displaysystem/`, `install/`, `interpreter/`...
+ - `_config.yml`: defines configuration options for docs website. See [jekyll/configuration](https://jekyllrb.com/docs/configuration/) for the other available config variables.
+ - `index.md`: the main page of `http://zeppelin.apache.org/docs/<ZEPPELIN_VERSION>/`
+
+ 
+### Markdown
+Zeppelin documentation pages are written with [Markdown](http://daringfireball.net/projects/markdown/). It is possible to use [GitHub flavored syntax](https://help.github.com/categories/writing-on-github/) and intermix plain HTML.
```
Comment (felixcheung) [inline contribution/documentation.md]: probably need to say Apache Zeppelin

File: contribution/documentation.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+---
+```
+
+ - `layout`: the default layout is `page` which is defined in `_layout/page.html`.
+ - `title`: the title for the document. Please note that if it needs to include `Zeppelin`, it should be `Apache Zeppelin`, not `Zeppelin`.
+ - `description`: a short description for the document. One or two sentences would be enough. This description also will be shown as an extract sentence when people search pages.
+ - `group`: a category of the document page
+
+### Headings
+All documents are structured with headings. From these headings, you can automatically generate a **Table of Contents**. There is a simple rule for Zeppelin docs headings.
```
Comment (felixcheung) [inline contribution/documentation.md]: Apache Zeppelin

File: contribution/general.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Contribution Guidelines
+
+**Zeppelin** is [Apache2 License](https://github.com/apache/zeppelin/blob/master/CONTRIBUTING.md) Software.
```
Comment (felixcheung) [inline contribution/general.md]: this and L24 - Apache Zeppelin

Comment (felixcheung) [inline contribution/general.md]: and throughout this file...

**Comment by corneadoug** [comment]
@Leemoonsoo 
We can reduce the font and width of the menu, but we can't do much more than that since the width of the documentation is fixed.

The sidemenu is inconsistent indeed, mainly because other categories might need more content.
One possible change would be to have:

* Contribution (general.html)
* Documentation (documentation.html)
* Webapplication (webapplication.html)

Showing as side menu for general.html, documentation.html

And to have:

* Zeppelin-web (webapplication.html)
* Defining Components (goodPracticeGuide01.html)

Showing as side menu for webapplication.html and goodPracticeGuide01.html

**Comment by corneadoug** [comment]
@felixcheung I don't think that adding Apache in front of every Zeppelin is necessary.
Especially since this is documentation for the project itself. It is fine to use the shorten name without Apache, and you can actually see that projects are doing it (Apache Mesos, Spark, Flink for example), and we've also been doing it.

**Comment by Leemoonsoo** [comment]
@corneadoug Because of width of the documentation is already fixed, I'm not sure it's best idea to reduce contents width even more.
Could you elaborate some possible alternatives to effectively display links of related pages in consistent way while maximizing actual content width?

File: contribution/general.md (hunk 3 above)
Comment (felixcheung) [inline contribution/general.md]: I'm pretty sure a title/subtitle/first word of the page needs to be "Apache project" http://www.apache.org/foundation/marks/pmcs.html#naming

**Comment by AhyoungRyu** [comment]
@felixcheung I agree with @corneadoug. If this document page need to be posted to the external resources, then we may need to attach `Apache` in front of every `Zeppelin`. But we don't need to say `Apache Zeppelin` every time in internal project page I guess :)

File: contribution/general.md (hunk 3 above)
Comment (AhyoungRyu) [inline contribution/general.md]: Yeah this Zeppelin should be `Apache Zeppelin` :)

**Comment by corneadoug** [comment]
@Leemoonsoo Let me try playing a bit more with the layout, I will post screenshots once its done

**Comment by corneadoug** [comment]
@Leemoonsoo I made both general.html and webapplication.html appear on the sideMenu.

For the size issue, I think the best would be to simply have the documentation max-width bigger (1024px instead of ~960px), I didn't push the CSS yet but the screenshot below have it.

If you Compare with the screenshots of the PR description, it gives more space to the content

screen shot 2016-08-30 at 1 24 25 pm

screen shot 2016-08-30 at 1 24 35 pm

**Comment by corneadoug** [comment]
I pushed the last CSS changes, and Updated the description's screenshots

**Comment by corneadoug** [comment]
@Leemoonsoo Merging if there is no more feedbacks

**Pull request closed without merging**